awesomerc.lua

```diff
@@ -95,10 +74,33 @@ mylauncher = awful.widget.launcher({ image = beautiful.awesome_icon,
 menubar.utils.terminal = terminal -- Set the terminal for applications that require it
 -- }}}
 
+-- {{{ Tag
+-- @DOC_LAYOUT@
+-- Table of layouts to cover with awful.layout.inc, order matters.
+tag.connect_signal("request::default_layouts", function()
```

Comment:
i can't understand what this warning means:
```
2020-01-12 02:17:37 W: awesome: `awful.layout.layouts` was set before `request::default_layouts` could be called. Please use `awful.layout.append_default_layouts` to  avoid this problem
```

when i do
```lua
tag.connect_signal("request::default_layouts", function()
  print("HI")
end)
```
nothing happens

--------------------------------

when i do
```
for _, l in ipairs(awful.layout.layouts) do
  awful.layout.remove_default_layout(l)
end
awful.layout.append_default_layouts(mylayoutslist)
```
it only adds layouts to default ones, but not replacing as `awful.layout.layouts = mylayoutlist` would do

Reply:
> nothing happens

I will investigate that, the signal is definitely sent when the first tag is created.

Overall, this change is a tradeoff. The problem with `awful.layout.layouts` is that it could only be set safely from one place. Setting it from multiple place required each setter to co-operate and use `table.insert`, which would break as soon as one setter just sets it. The overall goal, as we discussed a couple years ago, is that each API should be "module friendly" and can be called by modules rather than being only safe to use in `rc.lua`.

The use case for *this* one is a module which provide new layouts. Just by doing a `require("mylayoutmodule")`, you can add your layouts to the default set. 

> it only adds layouts to default ones, but not replacing as awful.layout.layouts = mylayoutlist would do

Should I add a `clear_layouts()` function? This would cover this use case. It is also mostly covered on the "tag object level" where you can set a list of layouts for a specific tag,

Reply:
that's actually weird what in default config it works, but not in mine:

when i replace both layouts and tags part from rc.lua i'm ending up with no tags at all, as neither `default_layouts` nor `desktop_decorations` signals are being called